### ISSUE TITLE
Missing symbol when building on gcc 7.1

### ISSUE THREAD

**Issue by pavanky** [arrayfire/arrayfire#1856]
Command used to reproduce:

```
mkdir build && cd build
cmake .. -DBUILD_CUDA=OFF -DBUILD_OPENCL=OFF -DBUILD_GRAPHICS=OFF && make example_helloworld_cpu
```

Error message:

```
[100%] Linking CXX executable helloworld/helloworld_cpu
../src/backend/cpu/libafcpu.so.3.5.0: undefined reference to `pthread_create'
collect2: error: ld returned 1 exit status
```

**Comment by mlloreda** [comment]
This seems to only occur when `-DBUILD_GRAPHICS=OFF`.

**Comment by pavanky** [comment]
@mlloreda This is also only happening when I use regular blas / cblas / lapack / lapacke packages (which are single threaded) instead of openblas (which is multi threaded). 

It's probably happening because the symbol is inherited from libraries that have pthread already linked to them.

**Comment by pavanky** [comment]
@umar456 do you think it is a good idea to link all liba* with thread libraries?

**Comment by umar456** [comment]
Fixed via #1872
